### DISCUSSION TITLE
scanner cycle setting on minio startup

### DISCUSSION THREAD

**Discussion by szabokacsa** [minio/minio#15727] [Q&A] [answered]
Hi,
is there a way (like env. variable) to configure (mc admin config set myminio scanner cycle=15s) the scanner cycle time on minio startup?

**Comment by vadmeste** [comment, marked as answer]
> Hi,
is there a way (like env. variable) to configure (mc admin config set myminio scanner cycle=15s) the scanner cycle time on minio startup?

It is 'MINIO_SCANNER_CYCLE', but normally you don't have to change the default value, are you facing any issue ?

**Reply by szabokacsa** [reply]
My original problen is that I want to speed up the frequesncy of the bucket lifecycle runs.
As I experienced the bucket lifecycle runs after each 16th scanner cycle.
I found this in the code:
![image](https://user-images.githubusercontent.com/3647735/191504992-7a332369-9583-45fd-94cb-39473328b070.png)

If the default scanner cycle time is 1min the bucket lifecycle runs in every 16 min, but it is not frequently enough.

If there is other way to increase the frequency of the bucket lifecycle running let me know.

**Reply by harshavardhana** [reply]
What is the point of running bucket lifecycle lesser than 16mins? Expiration objects happen only 24hrs+ at the minimum.

You have unnecessary expectations, if you want to do deletes frequently you can simply run `mc rm --versions --force --recursive` yourself in a cron job.

**Reply by szabokacsa** [reply]
We are running minio on a low profile system with limited disk space to store backup file which are uploaded frequently.
Also we are using bucket replication to sync data between mutiple system, so versioning is required.
The bucket lifecycle is deleting older versions of backup files.
Due to the frequently uploaded backup files we may run out of disk space if the bucket lifecycle runs on ever 16 min.

**Reply by klauspost** [reply]
@szabokacsa 

> If the default scanner cycle time is 1min the bucket lifecycle runs in every 16 min, but it is not frequently enough.

The 1 minute is time *between* cycles, so cycle time will be (time taken) + 1 minute.

You should add more space. There is no guarantees on lifecycle cycle time and you should leave plenty of margin.

**Comment by szabokacsa** [comment]
I set the scanner cycle time with MINIO_SCANNER_CYCLE, I is working fine.
Until we cannot increase storage space, this is a good solution for us.
The discussion can be closed. Thanks for your answers!
